### PR TITLE
BREAKING CHANGE: change TEST_KBC_PROJECTS environment to TEST_KBC_PROJECTS_FILE 

### PR DESCRIPTION
**Pull request by Matovidlo**
Jira: [PSGO-621](https://keboola.atlassian.net/browse/PSGO-621)

**Changes:**
- Load the projects from `build/ci/projects.json` locally or in CI. envsubstitute the projects and dump into `projects.json` within CI. 
- For local development the `projects.json` in are located in top level and the `TEST_KBC_PROJECTS_FILE` is absolute not relative. Update go-utils to newer version where `TEST_KBC_PROJECTS_FILE` is supported.

-----------

[PSGO-621]: https://keboola.atlassian.net/browse/PSGO-621?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: build/ci/projects.json [hunk 1]
```diff
@@ -0,0 +1,434 @@
+[
```
Comment (Matovidlo) [inline build/ci/projects.json:1]: Moved projects from actions env here.

File: docker-compose.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ services:
       - cache:/tmp/cache
     environment:
       # For all
-      - TEST_KBC_PROJECTS
+      - TEST_KBC_PROJECTS_FILE=/code/projects.json
```
Comment (Matovidlo) [inline docker-compose.yml:16]: default points to top level `projects.json`.

File: go.mod [hunk 3]
```diff
@@ -36,7 +36,7 @@ require (
 	github.com/joho/godotenv v1.5.1
 	github.com/jpillora/longestcommon v0.0.0-20161227235612-adb9d91ee629
 	github.com/keboola/go-client v1.26.4
-	github.com/keboola/go-utils v0.10.1
+	github.com/keboola/go-utils v0.10.2-0.20240521102004-ad17a99677bb
```
Comment (Matovidlo) [inline go.mod:39]: Utils that supports new environment variable that loads projects from `projects.json` file.

File: internal/pkg/service/cli/flag/flag.go [hunk 4]
```diff
@@ -5,14 +5,15 @@ import (
...
+	Help             configmap.Value[bool]   `configKey:"help" configShorthand:"h" configUsage:"print help for command"`
+	LogFile          configmap.Value[string] `configKey:"log-file" configShorthand:"l" configUsage:"path to a log file for details"`
+	LogFormat        configmap.Value[string] `configKey:"log-format" configUsage:"format of stdout and stderr"`
+	NonInteractive   configmap.Value[bool]   `configKey:"non-interactive" configUsage:"disable interactive dialogs"`
+	WorkingDir       configmap.Value[string] `configKey:"working-dir" configShorthand:"d" configUsage:"use other working directory"`
+	Verbose          configmap.Value[bool]   `configKey:"verbose" configShorthand:"v" configUsage:"print details"`
+	VerboseAPI       configmap.Value[bool]   `configKey:"verbose-api" configUsage:"log each API request and response"`
+	VersionCheck     configmap.Value[bool]   `configKey:"version-check" configUsage:"checks if there is a newer version of the CLI"`
+	TestProjectsFile configmap.Value[string] `configKey:"test-projects-file" configUsage:"file containing projects that could be used for templates"`
```
Comment (Matovidlo) [inline internal/pkg/service/cli/flag/flag.go:16]: new option that supports loading `projects.json` using flag not only `TEST_KBC_PROJECTS_FILE` env variable.

File: internal/pkg/template/repository/manager/repository.go [hunk 5]
```diff
@@ -107,7 +107,8 @@ func (r *CachedRepository) Template(ctx context.Context, reference model.Templat
 		r.d.Logger().Infof(ctx, `loading template "%s/%s"`, reference.FullName(), r.git.CommitHash())
 
 		// Load template
-		tmpl, err := loadTemplateOp.Run(ctx, r.d, r.repo, reference)
+		r.d.Logger().Infof(ctx, r.repo.Fs().BasePath())
```
Comment (Matovidlo) [inline internal/pkg/template/repository/manager/repository.go:110]: Should I use the BasePath or how the interaction with `test-projects-file` flag should work on repository level, vs template CLI Run level?

File: internal/pkg/template/template.go [hunk 6]
```diff
@@ -119,16 +119,17 @@ type _reference = model.TemplateRef
...
-	readme          string
-	manifestFile    *ManifestFile
-	inputs          StepsGroups
+	template         repository.TemplateRecord
+	version          repository.VersionRecord
+	deps             dependencies
+	fs               filesystem.Fs
+	srcDir           filesystem.Fs
+	testsDir         filesystem.Fs
+	projectsFilePath string
```
Comment (Matovidlo) [inline internal/pkg/template/template.go:128]: Add projectsFilePath to `template`.

**Review by michaljurecko — APPROVED** [review]
LGTM

**Pull request merged**